helpers/notification.js

```diff
@@ -0,0 +1,244 @@
+     * @param {object} res - an object
+     * @param {object} next - an object
+     * @return {boolean} Returns if true if it is valid else return false
+     * @static
+     */
+  static isValidOption(req, res, next) {
+    if (req.params.option === 'email') {
+      next();
+    } else if (req.params.option === 'in-app') {
+      next();
+    } else if (req.params.option === 'follower') {
+      next();
+    } else if (req.params.option === 'articleFavorite') {
+      next();
+    } else {
+      return res.status(422).send({ status: 422, Error: 'Only option must be email,in-app,follower or articleFavorite' });
+    }
+  }
+
+  /**
+   * @param {integer} authorId
+   * @returns {object} return list of followers
+   */
+  static async getFollowersList(authorId) {
+    const lists = await Follows.findAll({
+      where: { follower: authorId },
+      attributes: ['following']
+    });
+    return lists;
+  }
+
+  /**
+   * @param {object} userIds
+   * @param {string} userType
+   * @returns {Array} return array of object
+   */
+  static async serializeUsers(userIds, userType) {
+    const type = userType === 'follower' ? 'follower' : 'following';
+    const userList = [];
+    await Promise.all(userIds.map(async (currentId) => {
+      const userInfo = await passportHelper.findRecord(User, currentId[type]);
+      userList.push({
+        id: currentId[type],
+        username: userInfo.dataValues.username,
+        bio: userInfo.dataValues.bio,
+        image: userInfo.dataValues.image,
+        email: userInfo.dataValues.email,
+        notification: userInfo.dataValues.notification
+      });
+    }));
+    return userList;
+  }
+
+  /**
+   * @param {object} followers
+   * @returns {object} return emails
+   */
+  static getfollowersWithEmailOption(followers) {
+    const followersWithEmails = [];
+    followers.forEach((follower) => {
+      if (follower.notification.indexOf('email') !== -1) {
+        followersWithEmails.push(follower);
+      }
+    });
+    return followersWithEmails;
+  }
+
+  /**
+   * @param {object} followers
+   * @param {integer} authorId
+   * @param {string} articleSlug
+   * @returns {object} return followers
+   */
+  static getfollowersWithAppOption(followers) {
+    const followersWithApp = [];
+    followers.forEach((follower) => {
+      if (follower.notification.indexOf('in-app') !== -1) {
+        followersWithApp.push(follower);
+      }
+    });
+    return followersWithApp;
+  }
+
+  /**
+   *  Send notification email when users they follow publish new articles
+   * @param {string } email - A user aemail
+   * @param {string} author
+   * @param {string} title
+   * @returns {object} return email to send
+   */
+  static sendEmailToFollowers(email, author, title) {
+    return mailSender.send({
+      email,
+      subject: `${author} created a new article`,
+      html: `<html>You can check <strong>${title}</strong> created on Author Haven</html>`
+    });
+  }
+
```

Comment:
Why do you have these two similar functions `sendEmailToFollowers` and `sendEmailOnInteraction`? I suggest you create one function then you check if is interacted user or something else.

Reply:
Can you suggest how I can do it because email body and parameters are different

Reply:
> Why do you have these two similar functions `sendEmailToFollowers` and `sendEmailOnInteraction`? I suggest you create one function then you check if is interacted user or something else.

Call it like `sendEmailToUser` then pass `email`, `user`, and `title` like you did then check is user is `follower` or `interaction`, after set subject accordingly.

Reply:
I think with that option, it will be the same because It won't reduce the codes instead it will combine it and add conditional operators in one method. Unless  I am not getting it well